frontend/src/app/patients/ArchivePersonModal.tsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import { gql, useMutation } from "@apollo/client";
+import Modal from "react-modal";
+import Button from "../commonComponents/Button";
+import { displayFullName, showNotification } from "../utils";
+import "./ArchivePersonModal.scss";
+import { Patient } from "./ManagePatients";
+
+import Alert from "../commonComponents/Alert";
+import { toast } from "react-toastify";
+
+const ARCHIVE_PERSON_RECORD = gql`
+  mutation ArchivePerson($id: String!, $deleted: Boolean!) {
+    setPatientIsDeleted(id: $id, deleted: $deleted) {
+      internalId
+    }
+  }
+`;
+
+interface Props {
+  person: Patient;
+  closeModal: () => void;
+}
+interface ArchivePersonResponse {
+  internalId: string;
+}
+interface ArchivePersonParams {
+  id: string;
+  deleted: boolean;
+}
+const ArchivePersonModal = ({ person, closeModal }: Props) => {
+  const [archivePersonRecord] = useMutation<
+    ArchivePersonResponse,
+    ArchivePersonParams
+  >(ARCHIVE_PERSON_RECORD);
+  const yesArchivePerson = () => {
+    archivePersonRecord({
+      variables: {
+        id: person.internalId,
+        deleted: true,
+      },
+    })
+      .then(() => {
+        const alert = <Alert type="success" title="Record archived" body="" />;
+        showNotification(toast, alert);
+      })
+      .finally(() => {
```

Comment:
Should probably catch and handle errors here somehow, so we can at least report them to app insights

Reply:
Definitely could use a `.catch()` here.

Reply:
I'm using the utility that Tom wrote for reporting errors. It's kicking the problem to that utility but we need to figure out where and how that should be reporting them.